src/java.base/share/classes/java/lang/String.java

```diff
@@ -3129,41 +3129,50 @@ public String[] split(String regex, int limit) {
             (ch < Character.MIN_HIGH_SURROGATE ||
              ch > Character.MAX_LOW_SURROGATE))
```

Comment:
Not your change, but I'd replace them with `MIN_SURROGATE` and `MAX_SURROGATE` respectively, for readability, taking the opportunity.